### ISSUE TITLE
Parser does gets confused with __attribute__

### ISSUE THREAD

**Issue by ggnkua** [slynch8/10x#1438]
Consider the following snippet:

```#define GFXRAM   __attribute__ ((section (".gfx_data")))
#define MAXSCROLL 64*64
typedef struct {
    WORD	tile;
    WORD	attributes;
} Scroll3;

GFXRAM Scroll3 scroll3 [MAXSCROLL] __attribute__ ((aligned (0x10000))) = {};

int test(void)
{
    Scroll3 x = scroll3;
}```

`scroll3` isn't recognised as anything from the parser. But if in its definition we remove the macro `GFXRAM` and `__attribute__ ((aligned (0x10000)))` then it is working fine.

**Comment by slynch8** [comment]
done (0.35.95)

**Comment by ggnkua** [comment]
I'll have to bother you with this once more because I foolishly omitted something in the above code (namely, the definition of `GFXRAM`, and it's still triggering the same problem):

```
#define GFXRAM   __attribute__ ((section (".gfx_data")))
#define MAXSCROLL 64*64
typedef struct {
    WORD	tile;
    WORD	attributes;
} Scroll3;

GFXRAM Scroll3 scroll3 [MAXSCROLL] __attribute__ ((aligned (0x10000))) = {};

int test(void)
{
    Scroll3 x = scroll3;
}
```

So now there are two `__attribute__` in the same line, and 10x gets confused until `GFXRAM` is removed. So perhaps it isn't anticipating `__attribute__` at the start.

**Comment by slynch8** [comment]
no problem, easy fix.
fixed (0.35.99)
